mlrun/projects/project.py

```diff
@@ -1648,12 +1651,55 @@ def get_function(
             function = get_db_function(self, key)
             self.spec._function_objects[key] = function
         if enrich:
-            function = enrich_function_object(
-                self, function, copy_function=copy_function
+            function = self.enrich_function_object(
+                function, copy_function=copy_function
             )
             self.spec._function_objects[key] = function
         return function
 
+    def enrich_function_object(
+        self, function, decorator=None, copy_function=True
```

Comment:
Few notes regarding the method:
- I don't think this method should be exposed to the user.
- Also I don't think this method needs to be part of the project object but rather a helper of some kind.
- Missing docstring